libs/providers/flipt/src/lib/flipt-provider.ts

```diff
@@ -0,0 +1,228 @@
+import { AuthenticationStrategy, FliptClient } from '@flipt-io/flipt';
+import {
+  EvaluationContext,
+  Provider,
+  JsonValue,
+  ResolutionDetails,
+  ProviderStatus,
+  ProviderMetadata,
+  TypeMismatchError,
+  StandardResolutionReasons,
+  GeneralError,
+} from '@openfeature/server-sdk';
+import { transformContext } from './context-transformer';
+
+export interface FliptClientParameters {
+  url: string;
+  authenticationStrategy?: AuthenticationStrategy;
+}
+
+export class FliptProvider implements Provider {
+  readonly metadata: ProviderMetadata = {
+    name: 'flipt-client-provider',
+  };
+
+  private clientParameters: FliptClientParameters;
+  private namespace?: string;
+  private _client?: FliptClient;
+
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
+
+  constructor(namespace: string, clientParameters: FliptClientParameters) {
+    this.clientParameters = clientParameters;
+    this.namespace = namespace;
+  }
+
+  set status(status: ProviderStatus) {
+    this._status = status;
+  }
+
+  private set client(client: FliptClient) {
+    this._client = client;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  private get client(): FliptClient {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
+  }
+
+  async initialize(): Promise<void> {
+    this.client = new FliptClient({
+      url: this.clientParameters.url,
+      authenticationStrategy: this.clientParameters.authenticationStrategy,
+    });
+    this.status = ProviderStatus.READY;
+  }
+
+  async resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<boolean>> {
+    const evalContext: Record<string, string> = transformContext(context);
+
+    try {
+      const booleanEvaluation = await this._client?.evaluation.boolean({
```

Comment:
```suggestion
      const booleanEvaluation = await this.client.evaluation.boolean({
```

This will return or a client or throw.